lib/_http_server.js

```diff
@@ -477,20 +478,25 @@ function connectionListener(socket) {
       // if the user never called req.read(), and didn't pipe() or
       // .resume() or .on('data'), then we call req._dump() so that the
       // bytes will be pulled off the wire.
-      if (!req._consuming && !req._readableState.resumeScheduled)
+      if (!req._consuming && !req._readableState.resumeScheduled) {
         req._dump();
+      }
 
       res.detachSocket(socket);
+      res._last = outgoing.length === 0;
```

Comment:
We checking that on ```res:finish``` event, why it is not a right place to detect "is outgoing queue is empty?" In this place ```res._last``` inicates that we have no nore message to send.